plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -678,6 +682,45 @@ fu_genesys_tsdigit_value(gchar c)
 		return c - 'a' + 10;
 	return g_ascii_digit_value(c);
 }
+
+static gboolean
+fu_genesys_usbhub_device_get_bonding_and_flash_dump_location_bit(FuGenesysUsbhubDevice *self, GError **error)
+{
+	gint bonding;
+
+	if (self->dynamic_ts.bonding == 0)
+		return TRUE;
+
+	if (self->chip.model == ISP_MODEL_HUB_GL3590) {
+		self->bonding = (self->dynamic_ts.bonding & GL3590_BONDING_VALID_BIT);
+		if (self->dynamic_ts.bonding & GL3590_BONDING_FLASH_DUMP_LOCATION_BIT)
+			self->flash_dump_location_bit = TRUE;
```

Comment:
@hughsie which syntax is the best, that one or the following?

```
self->flash_dump_location_bit = self->dynamic_ts.bonding & GL3590_BONDING_FLASH_DUMP_LOCATION_BIT;
```

Reply:
I'd say the latter, as it sets to negative automatically if unset

Reply:
I agree.